src/hapi/pipelines/utilities/population.py

```diff
@@ -0,0 +1,90 @@
+"""Functions specific to the population theme."""
+
+import re
+from logging import getLogger
+from typing import Dict
+
+from sqlalchemy.orm import Session
+
+from hapi.pipelines.database.db_population import DBPopulation
+from hapi.pipelines.utilities.admins import Admins
+from hapi.pipelines.utilities.age_range import AgeRange
+from hapi.pipelines.utilities.gender import Gender
+from hapi.pipelines.utilities.metadata import Metadata
+
+logger = getLogger(__name__)
+
+
+def populate_population(
+    results: Dict,
+    session: Session,
+    metadata: Metadata,
+    admins: Admins,
+    gender: Gender,
+    age_range: AgeRange,
+):
+    logger.info("Populating population table")
+    for result in results:
+        resource_ref = metadata.data[result["resource"]["hdx_id"]]
+        reference_period_start = result["reference_period"]["startdate"]
+        reference_period_end = result["reference_period"]["enddate"]
+        for hxl_tag, values in zip(result["headers"][1], result["values"]):
+            if not _validate_hxl_tag(hxl_tag):
+                raise ValueError(f"HXL tag {hxl_tag} not in valid format")
+            gender_code, age_range_code = _get_hxl_mapping(hxl_tag=hxl_tag)
+            if gender_code is not None and gender_code not in gender.data:
+                raise ValueError(f"Gender code {gender_code} not in table")
+            if (
+                age_range_code is not None
+                and age_range_code not in age_range.data
+            ):
+                age_range.populate_single(age_range_code=age_range_code)
+            for admin_code, value in values.items():
+                population_row = DBPopulation(
+                    resource_ref=resource_ref,
+                    admin2_ref=admins.data[admin_code],
+                    gender_code=gender_code,
+                    age_range_code=age_range_code,
+                    population=value,
+                    reference_period_start=reference_period_start,
+                    reference_period_end=reference_period_end,
+                    # TODO: For v2+, add to scraper
+                    source_data="not yet implemented",
+                )
+
+                session.add(population_row)
+    session.commit()
+
+
+def _validate_hxl_tag(hxl_tag: str) -> bool:
+    # TODO: is this the right place to define the formats??
+    """Validate HXL tags
+
+    Assume they have the form:
```

Comment:
David once added hxl tags to population data for me, so I have his tags in [this spreadsheet](https://docs.google.com/spreadsheets/d/1HI62ioatXitVb56HphB4cVzEBjkabiOh/edit#gid=427852464) if they would be helpful!  I'm not sure if we should use +total or +all...

Reply:
Thanks Briar this is helpful. Do you think it's important we follow this guide? In the case of the pipeline, these hxl tags just used as an intermediate step to parse the data. Mostly I would maybe prefer to avoid including infant, adolescent, etc. which I think overcomplicates things.

Reply:
I agree - the age categories are definitely unnecessary.  I don't think you have to follow these exactly, just throwing it in in case it helps!

Reply:
Yes, interesting to see that instead of e.g. "80_plus" they just use "80_". I might stick with the plus for now since it's more explicit but we can always change.